store/state/value.go

```diff
@@ -0,0 +1,121 @@
+package state
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+)
+
+type Value struct {
+	base Base
+	key  []byte
+}
+
+func NewValue(base Base, key []byte) Value {
+	return Value{
+		base: base,
+		key:  key,
+	}
+}
+
+func (v Value) store(ctx Context) KVStore {
+	return v.base.store(ctx)
+}
+
+func (v Value) Cdc() *codec.Codec {
```

Comment:
Why do we need a getter for the codec? What if we passed in the codec instead of having it as part of `Value`?

Reply:
Updates on this suggestion @mossid ?

Reply:
Having codec as a field will make the code bit harder to read but okay with it, will change